server/bin/pbench-tarball-stats.py

```diff
@@ -91,12 +91,35 @@ def get_sat_name(ctrl: str):
     return "main" if len(parts) == 1 else parts[0]
 
 
-def transform_buckets(buckets: dict, sat_names: list):
+_sevendays = timedelta(days=7)
+
+
+def week_end(dt: datetime) -> str:
+    """Return a string representing the week ending date of the given datetime
+    object.
+    """
+    week_end = dt - timedelta(days=dt.isoweekday()) + _sevendays
+    return week_end.strftime("%Y-%m-%d")
+
+
+def transform_buckets(buckets: dict, sat_names: list, now: datetime, kind: str):
     """Transform a given bucket to calculate the % tar balls by the origin
     server.
     """
+    if kind == "y":
+        now_s = now.strftime("%Y")
+    elif kind == "m":
+        now_s = now.strftime("%Y-%m")
+    else:
+        assert kind == "w", f"error unexpected kind value, {kind!r}"
+        now_s = week_end(now)
+    twomonths = timedelta(days=60)
     new_buckets = {}
     for bucket_name, bucket_list in sorted(buckets.items()):
+        if kind == "w":
+            bucket_dt = datetime.strptime(bucket_name, "%Y-%m-%d")
+            if bucket_dt + twomonths < now:
+                continue
```

Comment:
This is a bit of a quibble, but...60 days never comprises two months (except when February has 29 days; otherwise, it's always 59, 61 or 62... 😉), and it's not an integral number of weeks, either.

Since what you're trying to do is limit the number of week-long buckets, I'd suggest just printing out the last 9 (or 10?) buckets.  E.g.:
```
    in_buckets = sorted(buckets.items())
    if kind == "w":
        in_buckets = in_buckets[-9:]
    for bucket_name, bucket_list in in_buckets:
```

Reply:
And, with suitable rearrangement, you can combine the `if` statement in my suggestion with the one at line 115.

Reply:
That works assuming all weeks have tar balls.  The method is to pick 60 days, and then find the "week ending" date of that time, so it ensures 8 or 9 weeks are used.

Reply:
OK.  From my perspective, the point is to limit the amount of output (in lines) in the report -- I'm not sure that I care how many days back it reaches -- so if there's a week without tarballs, that just means that there is an available spot for older data.

But, if you want to get all orthodox and reach back only a certain number of weeks, regardless of how many lines of output that produces, I s'pose that's fine, too.  (Although, the required code is a little more complicated.)